### DISCUSSION TITLE
Create Roast Queue for Efficient Production Roasting

### DISCUSSION THREAD

**Discussion by dadlerbean** [artisan-roaster-scope/artisan#1544] [Q&A] [answered]
I am new to Artisan, have question about production planning. Wondering if there is a way to create a queue of roasts for all the planned roasts for the day or week. The object is maximize efficiency and roast throughput, i.e. to avoid having stop to enter roasts one at a time. Hopefully my question is clear. Thanks for any advice/insight.

**Comment by poundy** [comment]
no, not very clear. Well perhaps my perception of where you could save any time that's worth saving is unclear?

I'm a small time home roaster and I usually roast 5-10 batches max, usually only on one day per month. Here's my process. 

If I'm organised, I will have a text file with my roast plan in it. That usually just lists coffee name, quantity, and number of batches ("Colombia XYZ, 2100, x3"). If I'm disorganised, I'll have my planning spreadsheet open in the background that has country of origin and quantity and batches needed.

I'll weigh my first few (or all) of my batches of green coffee out based on the plan while the machine is preheating... I only have 7 green buckets for my greens, so that determines if I have to go back and get more greens once roasting.

I have Artisan set to open the Roast Properties dialog box on CHARGE (the setting to do that is on that page, Ctrl-T is the shortcut to open it). So, I start a roast, load greens into the hopper, and when ready I'll CHARGE. From there, things happen relatively simply. Once the roaster is under control (airflow and gas set correctly etc), I'll go back to the PC. The dialog has popped up when I hit charge. For the first batch of a particular bean, I'll copy the coffee name from my text file (or look it up if I was disorganised) into the "beans" field in the roast properties page, and I'll tab into the green weight field and put the number in there.  If it's a second roast of the same bean and weight, nothing needs changing. The roast progresses and finishes. I have an automation to STOP the roast logging at 90 seconds past DROP, while I'm cooling the roast and resetting the roaster for BBP. I also have Artisan configured to load the completed roast into Viewer, so while I'm in BBP and not roasting I can weigh out the previous roast, jump over to viewer, Ctrl-T, 4x TABs to get into roasted weight field, and enter my number there, and save that profile. 

So all up, I don't think there's much time to be trimmed in that process?  Which means that there's not much to be gained... 

What area do you think you have that takes time ?

**Reply by dadlerbean** [reply]
HI Poundy, we have a pretty different situation, production roasting of about 30-50 batches per day, on a 12kg Diedrich - like to maintain an hourly throughput and keep the rhythm, roast dance, going. Stopping to enter roast details between every roast cycle would slow the pace significantly, and could be error-prone. (We are also testing Artisan.plus for inventory, which does help with roast entries and of course maintaining an inventory database). So, what we do now, with different software, is enter, and bucket, the roasts planned for the day during roaster warmup in the morning. Then we just roast continuously with minimal pauses between roasts. So the aim is to have all the planned roasts entered into a queue in advance and then just select from the created queue to match the bucket that is going into the roaster. Hopefully this is clearer than my question. I really appreciate your quick response, it helps my learning to know how others interact with Artisan. Thank you and have a good day!! David

**Comment by MAKOMO** [comment]
Roast scheduling is currently worked at. It will be introduced at WoC'24 in Copenhagen and available short after. This will take out that one step outlined above by @poundy of creating a written list of planned roasts and copying information from this list to the roasts properties.

**Reply by dadlerbean** [reply, marked as answer]
Hi Marko, this is great news. I look forward to trying this. Thanks for all the good work!! Best wishes, David
